tools/third_party.py

```diff
@@ -0,0 +1,164 @@
+    if os.name == 'nt' and not 'DEPOT_TOOLS_WIN_TOOLCHAIN' in env:
+        env['DEPOT_TOOLS_WIN_TOOLCHAIN'] = "0"
+    return env
+
+
+def fix_symlinks():
+    # Ensure the third_party directory exists.
+    try:
+        os.makedirs(third_party_path)
+    except:
+        pass
+
+    # Make symlinks to Yarn metadata living in the root repo.
+    remove_and_symlink("../package.json", tp("package.json"))
+    remove_and_symlink("../yarn.lock", tp("yarn.lock"))
+
+    # TODO(ry) Is it possible to remove these symlinks?
+    remove_and_symlink("v8/third_party/googletest", tp("googletest"), True)
+    remove_and_symlink("v8/third_party/jinja2", tp("jinja2"), True)
+    remove_and_symlink("v8/third_party/llvm-build", tp("llvm-build"), True)
+    remove_and_symlink("v8/third_party/markupsafe", tp("markupsafe"), True)
+
+    # On Windows, git doesn't create the right type of symlink if the symlink
+    # and it's target are in different repos. Here we fix the symlinks that exist
+    # in the root repo while their target is in the third_party repo.
+    remove_and_symlink("third_party/node_modules", root("node_modules"), True)
+    remove_and_symlink("third_party/v8/build", root("build"), True)
+    remove_and_symlink("third_party/v8/buildtools", root("buildtools"), True)
+    remove_and_symlink("third_party/v8/build_overrides",
+                       root("build_overrides"), True)
+    remove_and_symlink("third_party/v8/testing", root("testing"), True)
+
+
+# Run Yarn to install JavaScript dependencies.
+def run_yarn():
+    run(["yarn"], cwd=third_party_path)
+
+
+# Run Cargo to install Rust dependencies.
+def run_cargo():
+    # Deletes the cargo index lockfile; it appears that cargo itself doesn't do it.
+    # If the lockfile ends up in the git repo, it'll make cargo hang for everyone
+    # else who tries to run sync_third_party.
+    def delete_lockfile():
+        lockfiles = find_exts(
+            path.join(rust_crates_path, "registry/index"), '.cargo-index-lock')
+        for lockfile in lockfiles:
+            os.remove(lockfile)
+
+    # Delete the index lockfile in case someone accidentally checked it in.
+    delete_lockfile()
+
+    run(["cargo", "fetch", "--manifest-path=" + root("Cargo.toml")],
+        cwd=third_party_path,
+        merge_env={'CARGO_HOME': rust_crates_path})
+
+    # Delete the lockfile again so it doesn't end up in the git repo.
+    delete_lockfile()
+
+
+# Run gclient to install other dependencies.
+def run_gclient_sync():
+    # Depot_tools will normally try to self-update, which will fail because
+    # it's not checked out from it's own git repository; gclient will then try
+    # to fix things up and not succeed, and and we'll end up with a huge mess.
+    # To work around this, we rename the `depot_tools` directory to
+    # `{root_path}/depot_tools_temp` first, and we set DEPOT_TOOLS_UPDATE=0 in
+    # the environment so depot_tools doesn't attempt to self-update.
```

Comment:
Does setting `DEPOT_TOOLS_UPDATE=0` not prevent it from updating in the first place? 
This temp path renaming business is pretty nasty - we should avoid if possible...

Have you looked at `depot_tools/update_depot_tools_toggle.py`? If this file exists it will not auto update: `SENTINEL_PATH = os.path.join(DEPOT_TOOLS_ROOT, '.disable_auto_update')`